XenAdmin/SettingsPanels/GpuEditPage.cs

```diff
@@ -292,7 +292,9 @@ private void addButton_Click(object sender, EventArgs e)
                         return;
 
                     VGPU_type type = tuple.VgpuTypes[0];
-                    var vGpu = new VGPU();
+                    // Set vGPU device to VGPU_UNASSIGNED_DEVICE, does not assign any devcie ID to it.
+                    // When create vGPU, 0 is used to declare the first availabe slot in VM
+                    var vGpu = new VGPU (GpuAssignAction.VGPU_UNASSIGNED_DEVICE);

```

Comment:
When we create a new vGPU object, the device field is an empty string, I believe. Can we not simple use the empty string as an indication that there is no device ID assigned yet? I don't think we need an extra string for this.

Reply:
The device field is indeed initialised with `0`. If we want it empty string we should set it explicitly afterwards.

Reply:
Or it may be even simpler to set it to `null` so that afterwards the check in GpuAssignAction can be `vGpu.device ?? "0"`.

Reply:
```The device field is indeed initialised with 0. If we want it empty string we should set it explicitly afterwards```
No, it is initialised  with `VGPU_UNASSIGNED_DEVICE` now, which is showed as `-`. but it may be a good idea to assign it to `null` instead.

Reply:
No. When a class is initialised, its fields are initialised before the constructor is called. If the constructor sets a value on a field, this value overwrites the value the field has been initialised with. The device field is initialised as `private string _device = "0";` and the hyphen you assign to it overwrites the original value. But as discussed above, we don't need to add a new constructor, but rather set the property after we call `var vGpu = new VGPU();`.